### ISSUE TITLE
x86 backend hangs on Linux if there are real time process with higher priority

### ISSUE THREAD

**Issue by ivankochin** [open-mpi/hwloc#474]
### Details of the problem

If one of the cores is occupied by a process that was started using `chrt --rr 99 taskset <core_number> <executable>`  command, then the program hangs during the topology loading even if it was bind by numactl to a different cpuset. Setting `HWLOC_COMPONENTS=-x86` solves the problem, so it seems like this issue caused by the x86 backend. Also, this problem was successfully solved by using the `HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUMEMBINDING` topology flag unreleased feature (discussed in #366).

Since there is a workaround (`HWLOC_COMPONENTS=-x86` environment variable) and a solution that should be available in HWLOC 2.5 release, the issue was created just to define the current state of the problem. And I think that this issue may be closed after `HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUMEMBINDING` topology flag feature release.

### Reproducer:

while_true.cpp
```c++
int main() { while(true) {} }
```

hwloc_topology_loading_and_initialization.cpp
```c++
#include <hwloc.h>
#include <iostream>
#include <vector>
#include <thread>

void print_affinity(const hwloc_bitmap_t& bitmap) {
    char* buffer = new char[256];
    hwloc_bitmap_snprintf(buffer, 256, bitmap);
    printf("Affinity is: %s\n", buffer);
    delete[] buffer;
}

int main() {
    hwloc_topology_t topology;
    hwloc_topology_init(&topology);
    // if (hwloc_topology_set_flags(topology, HWLOC_TOPOLOGY_FLAG_IS_THISSYSTEM | HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUBINDING) != 0) { // SOLVES THE ISSUE BUT NOT AVAILABLE IN RELEASE PACKAGES
    //     printf("Wrong flags were set to the topology instance.");
    // }
    hwloc_topology_load(topology);

    hwloc_cpuset_t  process_cpu_affinity_mask  = hwloc_bitmap_dup(hwloc_topology_get_topology_cpuset (topology));
    hwloc_nodeset_t process_node_affinity_mask = hwloc_bitmap_dup(hwloc_topology_get_topology_nodeset(topology));
    printf("process cpu affinity: ");
    print_affinity(process_cpu_affinity_mask);
    printf("process nodes affinity: ");
    print_affinity(process_node_affinity_mask);
    hwloc_bitmap_free(process_node_affinity_mask);
    hwloc_bitmap_free(process_cpu_affinity_mask);
}

```

Compile these executables and run them from *different* terminals in the following way:

Terminal1:
```
chrt --rr 99 taskset <core_num> ./while_true.exe
```

Terminal2:
```
numactl -C <different_core_nums> ./hwloc_topology_loading_and_initialization.exe
```

**Comment by bgoglin** [comment]
Thanks for the info. I'll add this example in the doc of the new topology flags in 2.5(rc1 was released on monday, the final 2.5 will likely early next week).
Brice

**Comment by ivankochin** [comment]
The new `HWLOC_TOPOLOGY_FLAG_RESTRICT_TO_CPUBINDING` flag successfully solves the problem, so I close the issue.
